client/landing/stepper/hooks/use-flow-login-url.ts

```diff
@@ -0,0 +1,43 @@
+import { addQueryArgs } from '@wordpress/url';
+import { useLoginUrl } from '../utils/path';
+import { useFlowLocale } from './use-flow-locale';
+import type { Flow } from 'calypso/landing/stepper/declarative-flow/internals/types';
+
+type UseFlowLoginUrlProps = {
+	flow: Flow;
+	loginUrlParams?: Record< string, string | number >;
```

Comment:
Considering it is a hook specific to be used on flows, we can access the URL directly using the react-router the `useQuery`.

Reply:
Agreed that this shouldn't be on the flow. I am just not sure which query params should generally be passed on to login, and which ones should only be kept when redirecting back to the flow.

Reply:
And to be explicit, I think this would be a good follow-up change once we have the new hook in use across the flows.